pretyping/globEnv.ml

```diff
@@ -133,12 +133,25 @@ let hide_variable env expansion id =
   else
     env
 
-let protected_get_type_of env sigma c =
-  try Retyping.get_type_of ~lax:true env sigma c
-  with Retyping.RetypeError _ ->
+let rec check_named env sigma c =
+  match EConstr.kind sigma c with
+  | Var id -> ignore (Retyping.get_type_of ~lax:true env sigma c)
```

Comment:
Why don't you simply call Retyping.type_of_var? This is potentially more costly because you could have to reexpand an evar, and there are so many wrappers to retyping that it gets hard to understand exactly what we're checking.

Reply:
Because it wasn't exposed
I'm just going to put protected_get_type_of in retyping, I think it's nice enough there.